python/Ganga/Utility/execute.py

```diff
@@ -230,10 +228,10 @@ def execute(command,
 
     if update_env:
         env_output_key = 'env_output'
-        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key)
+        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key, require_output=True)
     if not shell:
         pkl_output_key = 'pkl_output'
-        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key)
+        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key, require_output=False)
```

Comment:
Just becasue we don't *require* output into the pickle stream in this case, doesn't mean that we don't *allow* it. The point is that if `require_output` is `True` and none is found then an error is printed. The converse if not the case, `require_output` being `False` and it finding output is absolutely fine and often expected.